lib/annotations2triannon/open_annotation.rb

```diff
@@ -261,6 +254,33 @@ def to_ttl
       @graph.dump(:ttl, standard_prefixes: true)
     end
 
+    private
+
+    # Sets the annotation ID as an RDF::URI from id, but if the id is nil, it
+    # will try to get the ID from the graph and, as a last resort, it will
+    # generate a unique ID using a UUID.
+    # @param id [UUID|URI|String] to identify an open annotation
+    def set_id(id)
+      raise ArgumentError, 'id cannot be an empty String' if (id.instance_of?(String) && id.empty?)
+      if id.nil?
+        # Try to get the ID from the graph or use a UUID
+        q = [nil, RDF.type, OA.Annotation]
+        @id = @graph.query(q).collect {|s| s.subject }.first || RDF::URI.parse(UUID.generate)
```

Comment:
Given that @azaroth42 said  "Triannon at some point should maintain those URIs (see provenance ticket) when they're supplied", I was thinking that perhaps there's a diff from DMS app having an assigned id (dunno when that happens) vs. pipeline code deciding to assign an id when it's missing  vs.  triannon being able to distinguish (in the future) between "this is an id I should pay attention to" and "I should assign a triannon id".  I believe the pipeline code assigning its own minted UUIDs  will make it hard for triannon to make that distinction, should we get that far in triannon implementation.   

From what I understand, the pair trees at the beginnings of the triannon ids help with fedora performance;  they are provided by fedora when I ask for a new container without specifying an id.

Again, I'm not going to insist on this change.  I think it should be made, but I won't insist.